src/main/java/com/google/edith/QueryItems.java

```diff
@@ -0,0 +1,151 @@
+package com.google.edith;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.users.UserService;
+import com.google.common.base.Splitter;
+import com.google.edith.servlets.Item;
+import com.google.edith.servlets.Receipt;
+import com.google.gson.Gson;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Finds the items on past receipts that have expired by the time of the user's next shopping trip.
+ */
+public class QueryItems {
```

Comment:
Consider making this class final

Reply:
the issue is I need to mock this class to test another class and i get a can't mock final class error - is there a way around that?

Reply:
You could create an interface and mock that (+ implement it here)